cli/internal/turbopath/turbopath.go

```diff
@@ -0,0 +1,68 @@
+// Package turbopath teaches the Go type system about six
+// different types of paths:
+// - AbsoluteSystemPath
+// - RelativeSystemPath
+// - AnchoredSystemPath
+// - AbsoluteUnixPath
+// - RelativeUnixPath
+// - AnchoredUnixPath
+//
+// Between these two things it is assumed that we will be able to
+// reasonably describe file paths being used within the system and
+// have the type system enforce correctness instead of relying upon
+// runtime code to accomplish the task.
+//
+// Absolute paths are, "absolute, including volume root." They are not
+// portable between System and Unix.
+//
+// Relative paths are simply arbitrary path segments using a particular
+// path delimiter. They are portable between System and Unix.
+//
+// Anchored paths are, "absolute, starting at a particular root."
```

Comment:
I think I nearly get the difference between anchored and relative paths. The part I am still having a little trouble with is what the use is for a relative path w/o an anchor. Maybe a path segment, like `.turbo` might be represented that way? Or is it that it can have leading `..` elements? It seems like `RelativeUnixPath` is used primarily as the return value from `memoizeGetTraversePath`. I would expect that to be anchored, since it's a relative path from turbo's repo root, but if it can traverse upwards, I think relative fits there?
 
`RelativeSystemPath` is I think just used to represent an individual file name.

FWIW, I'm fully on board w/ duplicate concrete implementations if in fact they are intended to denote different concepts.